### PR TITLE
Navigation implementation

### PR DESCRIPTION
**Pull request by Ne-anna**
In this pull request I have implemented navigation with drop-down, drop-down Menu un submenu. Cart can be closed by clicking on a cart icon. Menu and submenu can be closed by clicking on 'X' icon. All three components can be closed with outside click as well. 
Have modify data.ts file interfaces.

### CODE REVIEW

File: src/Components/Attoms/Hamburger-button/hamburger.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import "../../Attoms/Hamburger-button/hamburger.scss";
+
+const Haburger = () => {
```
Comment (ghost) [inline src/Components/Attoms/Hamburger-button/hamburger.tsx]: typo => Hamburger

File: src/Components/Attoms/Hamburger-button/hamburger.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import "../../Attoms/Hamburger-button/hamburger.scss";
```
Comment (ghost) [inline src/Components/Attoms/Hamburger-button/hamburger.tsx]: folder name => Atoms (typo)
generally, if possible try to follow a single pattern for folder names & component names
I see a mix of camelCase + snake-case + PascalCase :D

File: src/Components/Attoms/search-button/search-button.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { MenuItem } from "../../../data";
+import "../search-button/search-button.scss";
+
+const Search = (props: MenuItem) => {
```
Comment (ghost) [inline src/Components/Attoms/search-button/search-button.tsx]: optional, destructure

File: src/Components/Attoms/subMenu-back-button/submenu-back.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { BackButtonTitle } from "../../../data";
+import "../subMenu-back-button/submenu-back.scss";
```
Comment (ghost) [inline src/Components/Attoms/subMenu-back-button/submenu-back.tsx]: optional, currently kebab-case with camelCase mixed, pick one

File: src/Components/Molecules/cart-drop-down/cart-drop-down.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import "../cart-drop-down/cart-drop-down.scss";
+import { Cart } from "../../../data";
+
+const CartDropDown = (props: Cart) => {
+  return (
+    <div className="cart">
+      {props.cart.map((post, index) => {
```
Comment (ghost) [inline src/Components/Molecules/cart-drop-down/cart-drop-down.tsx:7]: avoid using index for keys where possible, if we don't have id, ignore
meant to to flag line 9, < div className="cart__item" key={index} >
especially for dynamic content

File: src/Components/Organisms/Navigation/navigation.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    setMainMenuOpen(false);
+  };
+
+  const closeSubMenu = () => {
+    setSubMenuOpen(false);
+    setMainMenuOpen(true);
+  };
+
+  useEffect(() => {
+    const closeWithOutsideClick = (e: any) => {
```
Comment (ghost) [inline src/Components/Organisms/Navigation/navigation.tsx]: avoid any

File: src/Components/Attoms/subMenu-back-button/submenu-back.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+const SubMenuBack = (props: BackButtonTitle) => {
```
Comment (ghost) [inline src/Components/Attoms/subMenu-back-button/submenu-back.tsx]: optional, destructure

File: src/Components/Molecules/sub-menu/sub-menu.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import "../sub-menu/sub-menu.scss";
+import { BackButtonTitle, MenuData } from "../../../data";
+import SubMenuBack from "../../Attoms/subMenu-back-button/submenu-back";
+
+const SubMenu = (props: MenuData & BackButtonTitle) => {
```
Comment (ghost) [inline src/Components/Molecules/sub-menu/sub-menu.tsx]: avoid defining component props with intersection type as the merged types can have conflict in properties which will resolve to `never` type
```
interface SubMenuProps {
menu: MenuItem[];
closeSubMenu: () => void;
title: string | undefined;
buttonBackPath: string;
buttonBackAltTag: string;
}
```

or for more robust approach
```
interface SubMenuProps {
menuData: MenuData;
backButton: BackButtonTitle
}
```

File: src/Components/Organisms/Navigation/navigation.tsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
+import "../Navigation/navigation.scss";
+import { NavigationData, Cart, MenuData, backButtonData } from "../../../data";
+import Haburger from "../../Attoms/Hamburger-button/hamburger";
+import CartDropDown from "../../Molecules/cart-drop-down/cart-drop-down";
+import Menu from "../../Molecules/menu/menu";
+import SubMenu from "../../Molecules/sub-menu/sub-menu";
+import { useRef, useState, useEffect } from "react";
+
+const Navigation = (
+  props: NavigationData & Cart & MenuData & backButtonData
```
Comment (ghost) [inline src/Components/Organisms/Navigation/navigation.tsx]: avoid intersection type as the subtypes will clash at one point

File: src/Components/Organisms/Navigation/navigation.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+) => {
+  const [isCartOpen, setCartOpen] = useState(false);
+  const [isMainMenuOpen, setMainMenuOpen] = useState(false);
+  const [isSubMenuOpen, setSubMenuOpen] = useState(false);
```
Comment (ghost) [inline src/Components/Organisms/Navigation/navigation.tsx]: optional, `useState` accepts generic type, can explicitly pass the type you want or let TS infer it
`const [isCartOpen, setCartOpen] = useState<boolean>(false);`

File: src/Components/Organisms/Navigation/navigation.tsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+          src={props.dividerPath}
+          alt={props.altTagDivider}
+        />
+        <div
+          className={`navigation__hamburger${
+            isMainMenuOpen ? " active" : isSubMenuOpen ? " active" : ""
+          }`}
+          onClick={toggleMenu}
+        >
+          <Haburger />
```
Comment (ghost) [inline src/Components/Organisms/Navigation/navigation.tsx]: typo => Hamburger

File: src/data.ts [hunk 12]
```diff
@@ -85,19 +97,7 @@ export interface FooterData {
...
-  cart: Cart[];
-  navigation: Navigation;
-  menu: Menu[];
-  backButtonTitle: BackButtonTitle;
-  carousel: Carousel[];
-  carouselButtons: CarouselButtons;
-  about: About;
-  gallery: GalleryData;
-  footer: FooterData;
-}
```
Comment (ghost) [inline src/data.ts]: why was this removed?
having this type helps you when you want to check the underlaying data structure
currently when you hover over cart property (line 101) in `defaultData` object you'll see that it's `never` which is not the case in your app :P

File: src/Components/Attoms/Hamburger-button/hamburger.tsx (hunk 2 above)
Comment (Ne-anna) [reply src/Components/Attoms/Hamburger-button/hamburger.tsx]: Based on your suggestion I have stick to snake-case pattern for folder names.

File: src/Components/Molecules/cart-drop-down/cart-drop-down.tsx (hunk 5 above)
Comment (Ne-anna) [reply src/Components/Molecules/cart-drop-down/cart-drop-down.tsx:7]: In original json file there was not given ID, so I had two option add it by myself in json file or use index as a key.

File: src/Components/Attoms/subMenu-back-button/submenu-back.tsx (hunk 7 above)
Comment (ghost) [reply src/Components/Attoms/subMenu-back-button/submenu-back.tsx]: like
const SubMenuBack = ({title, buttonBackPath, buttonBackAltTag}: BackButtonTitle) {
  //content
}

File: src/Components/Attoms/subMenu-back-button/submenu-back.tsx (hunk 4 above)
Comment (Ne-anna) [reply src/Components/Attoms/subMenu-back-button/submenu-back.tsx]: Changed to submenu-back.scss

**Review by Jalal999 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by VSlobolinskyi — APPROVED** [review]
